project-codeflare/codeflare-operator
# Check if workerGroupSpec is not empty before accessing it

# Issue link


Jira: https://issues.redhat.com/browse/RHOAIENG-6614 

# What changes have been made

- The failing Ray Tests were caused by an `index out of range` error in the webhook logic. The `workerGroupSpec` was attempted to be accessed when it doesn't exist.
- Check is now made to verify it's not empty before accessing it.


# Verification steps



0. Have all components deployed including this version of the CFO and Kueue.
1. Deploy KubeRay-Operator in the same namespace as referenced in the [DSCI `default-dsci`](https://github.com/project-codeflare/codeflare-operator/blob/a7619510bca4d70ed4362846703735f027106f10/pkg/controllers/raycluster_controller.go#L254-L256)
2. In the KubeRay repo, run the previously failing test:
`go test -timeout 30m -v ./test/e2e/rayjob_lightweight_test.go ./test/e2e/support.go`
3. Test should pass.


## Checks
- Testing Strategy
